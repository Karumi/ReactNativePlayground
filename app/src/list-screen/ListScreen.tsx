import * as React from "react";
import { FlatList, StyleSheet, Text, View } from "react-native";
import Toolbar from "../base-components/toolbar/Toolbar";
import translator from "../i18n";

export interface Props {
    navigation: any;
}
class ListScreen extends React.Component<Props> {
    public render() {
        return (
            <View style={styles.container}>
                <Toolbar
                    title={translator().listScreenTitle}
                    navigation={this.props.navigation}
                    showBackButton={true}
                />
                <FlatList
                    data={[
                        { key: "Devin" },
                        { key: "Jackson" },
                        { key: "James" },
                        { key: "Joel" },
                        { key: "John" },
                        { key: "Jillian" },
                        { key: "Jimmy" },
                        { key: "Julie" },
                        { key: "1 Devin" },
                        { key: "1 Jackson" },
                        { key: "1 James" },
                        { key: "1 Joel" },
                        { key: "1 John" },
                        { key: "1 Jillian" },
                        { key: "1 Jimmy" },
                        { key: "1 Julie" },
                        { key: "2 Devin" },
                        { key: "2 Jackson" },
                        { key: "2 James" },
                        { key: "2 Joel" },
                        { key: "2 John" },
                        { key: "2 Jillian" },
                        { key: "2 Jimmy" },
                        { key: "2 Julie" },
                        { key: "3 Devin" },
                        { key: "3 Jackson" },
                        { key: "3 James" },
                        { key: "3 Joel" },
                        { key: "3 John" },
                        { key: "3 Jillian" },
                        { key: "3 Jimmy" },
                        { key: "3 Julie" },
                    ]}
                    renderItem={({ item }) => <Text style={styles.item}>{item.key}</Text>}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
    },
    item: {
        padding: 10,
        fontSize: 18,
        height: 44,
    },
});
export default ListScreen;
