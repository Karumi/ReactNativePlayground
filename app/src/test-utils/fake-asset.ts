jest.mock("../assets/asset", () => {
return {
        default: (uri) => uri,
    };
});
