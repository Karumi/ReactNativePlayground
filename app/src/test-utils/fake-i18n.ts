
import { I18nResolver } from "i18n-ts";
import en from "../translations/en";

const i18n = { default: en };

jest.mock("../i18n", () => {
return {
        default: () => new I18nResolver(i18n,  "en").translation,
    };
} );
