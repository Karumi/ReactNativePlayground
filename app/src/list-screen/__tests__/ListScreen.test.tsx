import * as React from "react";
import * as renderer from "react-test-renderer";
import "../../test-utils/fake-i18n";
import ListScreen from "../ListScreen";

jest.mock("../../base-components/toolbar/Toolbar", () => {
    return {
        default: "Toolbar",
    };
});

it("renders correctly with defaults", () => {
    const screen = renderer.create(<ListScreen navigation={jest.fn()}/>).toJSON();
    expect(screen).toMatchSnapshot();
});
