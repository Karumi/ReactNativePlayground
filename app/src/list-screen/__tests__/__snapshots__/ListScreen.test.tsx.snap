// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with defaults 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <Toolbar
    navigation={[MockFunction]}
    showBackButton={true}
    title="List"
  />
</View>
`;
